service/core/action/policy/create.go

```diff
@@ -0,0 +1,121 @@
+package policy
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/factly/dega-server/config"
+	"github.com/factly/dega-server/service/core/model"
+	"github.com/factly/dega-server/util"
+	"github.com/factly/x/renderx"
+)
+
+func create(w http.ResponseWriter, r *http.Request) {
+	spaceID, err := util.GetSpace(r.Context())
+
+	if err != nil {
+		return
+	}
+
+	userID, err := util.GetUser(r.Context())
+
+	if err != nil {
+		return
+	}
+
+	space := &model.Space{}
+	space.ID = uint(spaceID)
+
+	err = config.DB.First(&space).Error
+
+	if err != nil {
+		return
+	}
+
+	uID := strconv.Itoa(userID)
+	oID := strconv.Itoa(space.OrganisationID)
+	sID := strconv.Itoa(spaceID)
+
+	policyReq := &policyReq{}
+
+	json.NewDecoder(r.Body).Decode(&policyReq)
+
+	ketoPolicy := &model.Policy{}
+
+	ketoPolicy.ID = "id:org:" + oID + ":app:dega:space:" + sID + ":" + policyReq.Name
+	ketoPolicy.Description = policyReq.Description
+	ketoPolicy.Effect = "allow"
+
```

Comment:
While merging multiple strings & int it is better to use **fmt.Sprintf("id:org:" ,space.OrganisationID, ":app:dega:space:", spaceID ,":", policyReq.Name)** instead of converting them to string

Reply:
This string is quite repetitive **"org:" + oID + ":app:dega:space:" + sID + ":"** I suggest to use something  like this 
commonString = fmt.Sprintf("org:", space.OrganisationID, ":app:dega:space:", spaceID , ":" )